www/js/tweakModel.js

```diff
@@ -0,0 +1,210 @@
+define([
+	"dojo/_base/declare", 
+	"dojo/_base/array",
+	"./equation"
+], function(declare, array, equation){
+	return declare(null,{
+
+		constructor: function(model_object){
+			this.curObj = model_object;
+			this.authObj = model_object.authorModelNodes;
+			this.studObj = model_object.studentModelNodes;
+		},
+		/*
+		copyParams: function(){
+			var getNewID = this.getStartID()+1;
+			var snodesCount = 0;
+			var rightStatus = {
+				description: {status: "correct", disabled: true},
+				qtyDescription: {status: "correct"},
+				units: {status: "correct", disabled: true},
+				value: {status: "correct", disabled: true},
+				variable: {status: "correct", disabled: true},
+				variableType: {status: "correct", disabled: true}
+			}
+			//traverse through authObj
+			for(var prop in this.authObj){
+				//console.log("current prop is", prop, this.authObj.hasOwnProperty(prop));
+				if(this.authObj.hasOwnProperty(prop) && this.authObj[prop].hasOwnProperty("type") && this.authObj[prop].type == "quantity"){
+					var curAuthID = this.authObj[prop].ID;
+					var parentDet = this.findParentDetails(curAuthID);
+					this.curObj["studentModelNodes"][snodesCount++] = {
+						ID: "id" + getNewID++,
+						attemptCount: {
+							schema: 1,
+							entity: 0,
+							variables: 0,
+						},
+						authoredID: curAuthID,
+						color: this.authObj[prop].color,
+						description: this.authObj[prop].description,
+						links: this.authObj[prop].links,
+						parentSchema: parentDet.schema,
+						parentEquation: parentDet.equation,
+						position: this.authObj[prop].position,
+						selfEquation: this.authObj[prop].selfEquation,
+						status: rightStatus,
+						type: "quantity",
+						units: this.authObj[prop].units,
+						value: this.authObj[prop].value,
+						variable: this.authObj[prop].variable,
+						variableType: this.authObj[prop].variableType
+					}
+					this.updateAssistanceScore(curAuthID); 
+				}
+			}
+		},
+		*/
+
+		updateModel: function(showParams, showSchema){
```

Comment:
I like that the showSchema part can be easily changed to use different schema later if we want, that's good.

Show is confusing here, as it used for show/hide.  I think we should rename it to "give" everywhere, including the URL parameters (changed them from "ss/sp" to "gs/gp").


```suggestion
		updateModel: function(giveParams, giveSchema){
```